Added homework for the fourteenth lessons

AlexandrKarpilovich opened the pull request:
Добавлен 14й урок в файл README.md и домашняя работа по нему, но только обязательная часть

Сложности: 
Не до конца разобрался в работе wait() и notify()

AlexandrKarpilovich committed:
Added homework for the fourteenth lessons

artempolozov commented on src/lesson_14_Java/ThreadFirstHomeworkMain.java:
```diff
@@ -0,0 +1,33 @@
+        HelloTreadClass hello = new HelloTreadClass();
+
+        // Loop through from 0 to 20 and create threads
+        for (int i = 0; i < 20; i++) {
```
поток main ничего не знает о циклах. он только запускает потоки. циклы должны быть в потоках
задание выполнено неверно

artempolozov reviewed (commented):
второе задание не выполнено
первое - выполнено неверно
оценка 6 из 10

AlexandrKarpilovich committed:
Improved solutions to tasks

This topic is very difficult to come by. I will continue to understand myself with multithreading

artempolozov commented on src/lesson_14_Java/FirstDecision/MyThread.java:
```diff
@@ -0,0 +1,27 @@
+    @Override
+    public void run() {
+        synchronized (this) {
+            for (int i = 0; i < 20; i++) {
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+                System.out.println(this.phrase);
+            }
+        }
```
задание выполнено неверно. потоки не последовательны
в нескольких случаях из 10 они будут идти друг за другом. в остальных случаях - нет

artempolozov commented on src/lesson_14_Java/SecondDecision/MyThread.java:
```diff
@@ -0,0 +1,28 @@
+    @Override
+    public void run() {
+        synchronized (this) {
+            for (int i = 0; i < 20; i++) {
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+                System.out.println(this.phrase);
+            }
+            notify();
```
то же самое

artempolozov reviewed (commented):
оценка не изменилась
